File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/FloatInputFieldView.java
package com.saizad.mvvm.components.form.ui;

import android.content.Context;
import android.util.AttributeSet;

import com.sa.easyandroidform.ObjectUtils;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class FloatInputFieldView extends InputFieldView<Float> {

    public FloatInputFieldView(@NotNull Context context) {
        super(context);
    }

    public FloatInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public FloatInputFieldView(@NotNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public Float resolveFrom(@NotNull CharSequence charSequence) {
        try {
            return Float.valueOf(charSequence.toString());
        }catch (NumberFormatException e){
            return null;
        }
    }

    @Override
    public CharSequence resolve(@Nullable Float value) {
        if(ObjectUtils.isNull(value)){
            return null;
        }
        return value.toString();
    }

    @Override
    public boolean isSame(Float value, Float prevValue) {
        return value.equals(prevValue);
    }
}

File: saizad-mvvm/src/main/java/com/saizad/mvvm/components/form/ui/FloatInputFieldView.kt
package com.saizad.mvvm.components.form.ui

import android.content.Context
import android.util.AttributeSet
import com.sa.easyandroidform.ObjectUtils

open class FloatInputFieldView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : InputFieldView<Float>(context, attrs, defStyleAttr) {


    override fun resolveFrom(charSequence: CharSequence): Float? {
        return try {
            java.lang.Float.valueOf(charSequence.toString())
        } catch (e: NumberFormatException) {
            null
        }
    }

    override fun resolve(value: Float?): CharSequence? {
        return if (ObjectUtils.isNull(value)) {
            null
        } else value.toString()
    }

    override fun isSame(value: Float, prevValue: Float): Boolean {
        return value == prevValue
    }
}